go/private/go_mod.bzl

```diff
@@ -0,0 +1,151 @@
+# Copyright 2025 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+def version_from_go_mod(module_ctx, go_mod_label):
+    """Returns a version string from a go.mod file.
+
+    Args:
+       module_ctx: a https://bazel.build/rules/lib/module_ctx object passed
+            from the MODULE.bazel call.
+       go_mod_label: a Label for a `go.mod` file.
+
+    Returns:
+      a string containing the version of the Go SDK defined in go.mod
+    """
+    _check_go_mod_name(go_mod_label.name)
+    go_mod_path = module_ctx.path(go_mod_label)
+    go_mod_content = module_ctx.read(go_mod_path)
+
+    state = {
+        "toolchain": None,
+        "go": None,
+    }
+
+    current_directive = None
+    for line_no, line in enumerate(go_mod_content.splitlines(), 1):
+        tokens, _ = _tokenize_line(line, go_mod_path, line_no)
+        if not tokens:
+            continue
+
+        if not current_directive:
+            if tokens[0] not in ["module", "go", "require", "replace", "exclude", "retract", "toolchain"]:
```

Comment:
Since we only care about a very minor part of the file, I could see us ignoring unknown lines (i.e., deleting this check). Let's give that a try, mostly to avoid the requirement to update rules_go when a new version of Go comes out (in fact, we are already missing support for the `tool` directive).

Reply:
Done